### PR TITLE
Dynamic linking, Skipping stacks, and bump pathparams & queryparams to 2.0.2 

### PR DESCRIPTION
**Pull request by theweiweiway**
2 new snippets & 2 updated snippets

@Milad-Akarie for review

@johnpryan to address your comment about BookDetailsPage having no back button for the pathparams snippet - I'm pushing the route now so it has one now

Also, queryparam scenario i return it to `replace` since @chunhtai saying that it should replace, not push. However, I see that other libraries and pushing instead of replacing so I'm a bit confused on that.

Auth & Nesting are almost done but need Milad's input - we are working on those together and will be done very soon - sorry for cutting it this close

### CODE REVIEW

File: nav2-usability/scenario_code/lib/deeplink-pathparam/deeplink_pathparam_auto_route.dart [hunk 1]
```diff
@@ -65,7 +65,7 @@ class _BooksListScreenState extends State<BooksListScreen> {
               title: Text(book.title),
               subtitle: Text(book.author),
               onTap: () =>
-                  context.router.replace(BookDetailsRoute(id: books.indexOf(book))),
+                  context.router.pushNamed("/book/${books.indexOf(book)}"),
```
Comment (chunhtai) [inline nav2-usability/scenario_code/lib/deeplink-pathparam/deeplink_pathparam_auto_route.dart:68]: If user deeplink to this page, will they see the backbutton?

**Review by chunhtai — COMMENTED** [review]
I remembered that I asked some other package authors about them using pushNamed. Their answer was that their push is actually replacing the current page, it was just that their naming is a bit confusing.

File: nav2-usability/scenario_code/lib/deeplink-pathparam/deeplink_pathparam_auto_route.dart (hunk 1 above)
Comment (johnpryan) [reply nav2-usability/scenario_code/lib/deeplink-pathparam/deeplink_pathparam_auto_route.dart:68]: Yes I think we would like this to display an in-app back button to match the other samples

Comment (theweiweiway) [reply nav2-usability/scenario_code/lib/deeplink-pathparam/deeplink_pathparam_auto_route.dart:68]: Whoops, my bad - just added it

File: nav2-usability/scenario_code/lib/deeplink-pathparam/deeplink_pathparam_auto_route.dart [hunk 2]
```diff
@@ -43,7 +43,7 @@ class BooksApp extends StatelessWidget {
 
     return MaterialApp.router(
       routerDelegate: _appRouter.delegate(),
-      routeInformationParser: _appRouter.defaultRouteParser(),
+      routeInformationParser: _appRouter.defaultRouteParser(includePrefixMatches: true),
```
Comment (chunhtai) [inline nav2-usability/scenario_code/lib/deeplink-pathparam/deeplink_pathparam_auto_route.dart:46]: Just curious, can this only apply to a specific route? e.g. I want prefix matches for /books/123, but not for /user/123?

File: nav2-usability/scenario_code/lib/skipping-stacks/skipping_stacks_auto_route.dart [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+final AppState appState = AppState();
+
+class BooksApp extends StatelessWidget {
+  final _appRouter = AppRouter();
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp.router(
+      routeInformationParser: _appRouter.defaultRouteParser(),
+      routerDelegate: _appRouter.delegate(),
```
Comment (chunhtai) [inline nav2-usability/scenario_code/lib/skipping-stacks/skipping_stacks_auto_route.dart:72]: We also need the same prefix match here

File: nav2-usability/scenario_code/lib/deeplink-pathparam/deeplink_pathparam_auto_route.dart (hunk 2 above)
Comment (theweiweiway) [reply nav2-usability/scenario_code/lib/deeplink-pathparam/deeplink_pathparam_auto_route.dart:46]: I think you would have to use a RouteGuard in that case and push the specific routes you want

File: nav2-usability/scenario_code/lib/skipping-stacks/skipping_stacks_auto_route.dart (hunk 3 above)
Comment (theweiweiway) [reply nav2-usability/scenario_code/lib/skipping-stacks/skipping_stacks_auto_route.dart:72]: Done!~~~!

**Review by chunhtai — APPROVED** [review]
LGTM

**Pull request merged**